python/orca/src/bigdl/orca/learn/tf2/pyspark_estimator.py

```diff
@@ -372,29 +379,39 @@ def predict(self, data, batch_size=None, verbose=1,
             data_config=data_config
         )
 
+        def transform_func(iter, init_param, param):
+            partition_data = list(iter)
+            # res = combine_in_partition(partition_data)
+            param["data_creator"] = make_data_creator(partition_data)
+            return SparkRunner(**init_param).predict(**param)
+
         if isinstance(data, DataFrame):
-            pre_predict_data = data.repartition(self.num_workers)
```

Comment:
repartition is removed as it is not needed for predict. Also for the result data, the user would expect the partition to be unchanged.

Reply:
If we removed repartition, if the original partition number is big( like 2000 partitions), on each partition it will create a model to do predict. So there will be 2000 models created. If the model is big, there may cause memory issue. I think we can change to coalesce to reduce shuffling.

Reply:
On spark, the task running in the same time equals to the total cores,  Model will be GCed after the partition is finished. So there won't be 2000 models in the same time.

Reply:
even there are number of cores of models are running at the same time, if the model is big, there may have memory issue. @qiuxin2012 , is it right?

Reply:
@jenniew Yes, so user should ensure they have enough memory matching their executor cores, or they should decrease the core number.